File: commonlib/StreamData.cs
using System;

namespace commonlib
{
    public class StreamData
    {
        public string Filename { get; set; }
        public long Size { get; set; }
        public long DurationMS { get; set; }
        public int? FramesPerSecond { get; set; }
        public int? Width { get; set; }
        public int? Height { get; set; }
        public string Codec { get; set; }

        public override string ToString()
        {
            return $"StreamData: Filename '{Filename}', Size {Size}, Duration {DurationMS}, FPS ({FramesPerSecond}), WidthxHeight {Width}x{Height}, Codec '{Codec}'";
        }
    }
}

File: camsimweb/src/app/shared/interfaces/streamdata.ts
export interface IStreamData {
  Filename: string;
  Size: number;
  DurationMS: number | undefined;
  FramesPerSecond: number | undefined;
  Width: number | undefined;
  Height: number | undefined;
  Codec: string;
}

// export class StreamData {
//   Filename: string;
//   Size: number;
//   DurationMS: number | undefined;
//   FramesPerSecond: number | undefined;
//   Width: number | undefined;
//   Height: number | undefined;
//   Codec: string;

//   constructor(streamData: IStreamData) {
//     this.Filename = streamData.Filename;
//     this.Size = streamData.Size;
//     this.DurationMS = streamData.DurationMS;
//     this.FramesPerSecond = streamData.FramesPerSecond;
//     this.Width = streamData.Width;
//     this.Height = streamData.Height;
//     this.Codec = streamData.Codec;
//   }
// }
